Realm/RLMArray.mm

```diff
@@ -437,3 +438,23 @@ - (instancetype)reversedSortDescriptor {
 }
 
 @end
+
+@interface RLMArray (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArray (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    @throw RLMException(@"Unexpected handover of unmanaged `RLMArray`");
```

Comment:
Does this happen due to a bug in our code, or due to API misuse? If it's the former it should be an assertion, and if it's the latter the error message needs to do a better job of explaining to the user what they did wrong.